src/scalarstats.jl

```diff
@@ -37,6 +37,25 @@ function harmmean(a::RealArray)
     return n / s
 end
 
+# Generalized mean
+"""
+    genmean(a, p)
+
+Return the generalized/power mean with exponent `p` of a real-valued array.
+i.e. returns ((a_1+a_2+...+a_n)/n)^(1/p), where n is the length of the array a.
+"""
+function genmean(a::RealArray, p::Real)
+  if p == 0
+    return geomean(a)
+  end
+  s = 0.0
+  n = length(a)
+  for x in a
+    @inbounds s += convert(Float64, x)^p
```

Comment:
Makes sense. Please keep the comment about the domain error with integers, though. Else it's hard to understand the justification for this conversion.